src/components/VotesComp.jsx

```diff
@@ -1,7 +1,50 @@
-export const VotesComp = ({ article: { votes } }) => {
+import { useState } from 'react';
+import { patchVotesByArticleId } from '../utils/api';
+
+export const VotesComp = ({ article: { article_id, votes } }) => {
+  const [voteClicks, setVoteClicks] = useState(0);
+  const [upVoteClicked, setUpVoteClicked] = useState(false);
+  const [downVoteClicked, setDownVoteClicked] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  function handleClick(num) {
+    setVoteClicks((curr) => {
+      return curr + num;
+    });
+
+    patchVotesByArticleId(article_id, num)
+      .then((res) => {
+        console.log(res); //<--i will remove, but thought i should leave in so you could see it working
+      })
+      .catch((err) => {
+        setVoteClicks((curr) => {
+          return curr - num;
+        });
+      });
+  }
+
```

Comment:
I should be able to see if the vote has not worked - via something appearing - whether that's a modal saying 'sorry vote unsuccessful' or something along those lines!

Reply:
oh i didn't include the setstate function in the catch block. i had set something up if the response from the server was an error, but linked the two!